driver/task.go

```diff
@@ -226,21 +224,6 @@ func (t *Task) Source() string {
 	return t.source
 }
 
-// UserDefinedMeta returns metadata grouped by service
-func (t *Task) UserDefinedMeta() map[string]map[string]string {
-	t.mu.RLock()
-	defer t.mu.RUnlock()
-
-	meta := make(map[string]map[string]string)
-	for k, nestedMap := range t.userDefinedMeta {
-		meta[k] = make(map[string]string)
-		for nestedK, nestedV := range nestedMap {
-			meta[k][nestedK] = nestedV
-		}
-	}
```

Comment:
Now that user-defined-meta is a field on services, does this type of 'copying' logic have to be added to the Services() getter, or maybe it's ok as is?

Reply:
Believe so since maps are by reference, good catch.

Reply:
Love me some go playground https://play.golang.org/p/L-KljbPqTqZ